lib/appsignal.rb

```diff
@@ -171,13 +176,23 @@ def _load_config!(env_param = nil)
           Appsignal::Utils::StdoutAndLoggerMessage.warning(message)
         else
           # Load it when no config is present
-          load_dsl_config_file(context.dsl_config_file, env_param)
+          #
+          # We don't pass the `env_var` or `context.env` here so that the
+          # `Appsignal.configure` or `Appsignal.start` can figure out the
+          # environment themselves if it was not explicitly set.
+          # This way we prevent forcing an environment that's auto loaded on
+          # the to-be-loaded config file.
+          #
+          # The `(proper)_env_param` is only set when it's explicitly set,
+          # which means it needs to override any auto detected environment.
+          load_dsl_config_file(context.dsl_config_file, proper_env_param)
         end
       else
         # Load config if no config file was found and no config is present yet
         # This will load the config/appsignal.yml file automatically
         @config ||= Config.new(context.root_path, context.env)
       end
+      block.call(config) if block_given?
```

Comment:
Would it make sense to add a comment here explaining why this exists?

Reply:
Future readers may thank us :)